gapir/cc/resource_cache.cpp

```diff
@@ -25,21 +28,21 @@ ResourceCache::ResourceCache(std::unique_ptr<ResourceProvider> fallbackProvider)
 
 bool ResourceCache::get(const Resource*         resources,
                         size_t                  count,
-                        const ServerConnection& server,
+                        ReplayConnection*       conn,
                         void*                   target,
                         size_t                  size) {
     uint8_t* dst = reinterpret_cast<uint8_t*>(target);
     Batch batch(dst, size);
     for (size_t i = 0; i < count; i++) {
         const Resource& resource = resources[i];
         if (size < resource.size) {
-            return false; // Not enough space
+            return false;  // Not enough space
         }
         // Try fetching the resource from the cache.
         if (getCache(resource, dst)) {
             // In cache. Flush the pending requests.
-            if (!batch.flush(*this, server)) {
-                return false; // Failed to get resources from fallback provider.
+            if (!batch.flush(*this, conn)) {
```

Comment:
Note: This implementation can result in many round trips to the server, and its design was based around the idea that we could load the resource directly into the destination buffer without temporary copies. Given that gRPC forces us to have temporary copies, this implementation should be changed to have a single fetch.
Not suggesting this should be implemented in this PR, but a TODO would be nice.